admin/src/main/scala/io/buoyant/admin/names/DelegateApiHandler.scala

```diff
@@ -169,29 +175,40 @@ object DelegateApiHandler {
     def writeBuf[T](t: T): Buf = Buf.ByteArray.Owned(mapper.writeValueAsBytes(t))
   }
 
-  def getDelegateRsp(dtab: String, path: String, interpreter: NameInterpreter): Future[Response] =
+  def getDelegateRsp(dtab: String, path: String, delegator: Delegator): Future[Response] =
     (dtab, path) match {
       case (DtabStr(d), PathStr(p)) =>
-        Delegator(d, p, interpreter).values.toFuture().flatMap(Future.const).map { tree =>
-          val rsp = Response()
-          rsp.content = Codec.writeBuf(tree)
-          rsp.contentType = MediaType.Json
-          rsp
-        }
+        delegator.delegate(d, p).values.toFuture()
+          .flatMap(Future.const)
+          .flatMap(JsonDelegateTree.mk).map { tree =>
+            val rsp = Response()
+            rsp.content = Codec.writeBuf(tree)
+            rsp.contentType = MediaType.Json
+            rsp
+          }
       case _ => err(Status.BadRequest)
     }
 }
 
 class DelegateApiHandler(
-  namers: Seq[(Path, Namer)]
+  interpreters: String => NameInterpreter
```

Comment:
Sure, it just moves where you have to handle the case where the interpreter is not a delegator.  Having that handled here means we can handle it in a consistent way.